src/classes/BGE_DataImportBatchEntry_CTRL.cls

```diff
@@ -127,62 +127,108 @@ public with sharing class BGE_DataImportBatchEntry_CTRL {
     }
 
     /*******************************************************************************************************
-    * @description Retrieves all open Opportunities and unpaid Payments for the selected Donor (Account or Contact)
-    * @return JSON stringified Map<String, List<SObject>> of all openOpportunities and unpaidPayments for the given donor
+    * @description Retrieves all open Opportunities and related Payments for the selected Donor (Account or Contact)
+    * @return JSON string of all open Opportunities and related Payments for the given donor
     */
     @AuraEnabled
-    public static String getOpenDonations(Id donorId, String donorType) {
+    public static String getOpportunitiesWithOppPayments(Id donorId) {
+        Schema.SObjectType token = donorId.getSobjectType();
+
         String relationshipField;
-        if (donorType == 'Account1') {
+        if (token == Account.SObjectType) {
             relationshipField = 'AccountId';
-        } else if (donorType == 'Contact1') {
+        } else if (token == Contact.SObjectType) {
             relationshipField = 'Primary_Contact__c';
+        } else {
+            throw new AuraHandledException('SObjectType ' + token + ' is not supported.');
```

Comment:
This else is super unlikely to happen and so we don't really need a custom label, right? I sifted through our custom labels and didn't find anything more generic around _Something something_ + ' is not supported.' Should we?